hazelcast/src/main/java/com/hazelcast/datastore/DataStoreSupplier.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.datastore;
+
+import java.util.function.Supplier;
+
+public interface DataStoreSupplier<DS> extends Supplier<DS>, AutoCloseable {
```

Comment:
a javaDoc for the end-user to concisely tell how this method should be used.

Reply:
Fixed: https://github.com/hazelcast/hazelcast/pull/22372/commits/affa1b4901b6f360764f92ebc26b68157367651b

Reply:
Fixed: https://github.com/hazelcast/hazelcast/pull/22372/commits/affa1b4901b6f360764f92ebc26b68157367651b

Reply:
Fixed: https://github.com/hazelcast/hazelcast/pull/22372/commits/affa1b4901b6f360764f92ebc26b68157367651b